File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/fight/character/GameFightRefreshFighterMessage.scala





















// Generated on 05/03/2013 10:34:41
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameFightRefreshFighterMessage extends MessageDeserializer {
  val messageId = 6309

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameFightRefreshFighterMessage.messageId

    val informations: GameContextActorInformations.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeUShort(informations.typeId)
      informations.serialize(buf)
    }

  }

  protected def create(informations: GameContextActorInformations.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val informations = types.get[GameContextActorInformations.TypeImpl](buf.readShort());

    create(informations)
  }

}

object GameFightRefreshFighterMessage extends GameFightRefreshFighterMessage {
  case class MessageImpl(informations: GameContextActorInformations.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(informations: GameContextActorInformations.TypeImpl) = MessageImpl(informations)

  def apply(informations: GameContextActorInformations.TypeImpl) = create(informations)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameFightRefreshFighterMessage.java


// Generated on 05/08/2013 19:37:44
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightRefreshFighterMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6309;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public GameContextActorInformations informations;
    
    public GameFightRefreshFighterMessage() { }
    
    public GameFightRefreshFighterMessage(GameContextActorInformations informations) {
        this.informations = informations;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(informations.getTypeId());
        informations.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        informations = ProtocolTypeManager.getInstance().build(buf.readShort());
        informations.deserialize(buf);
    }
    
}
